### PR TITLE
The checked property is ignored on checkbox columns where the field has a value

### PR DESCRIPTION
**Pull request by TantumLabs**
The checked property is ignored if the field has a value.

For example:
I am setting the checkbox column to the Id field so there is always a value. If I set the formatter to checked: false the checkbox is still checked because the field has a value.

`function checkboxFormatter(){
return {
checked: false,
disabled: true
}`

**Bug fix?**
yes

**New Feature?**
no

**Resolve an issue?**

**Example(s)?**
https://live.bootstrap-table.com/code/TantumLabs/4276

### CODE REVIEW

File: src/utils/index.js [hunk 1]
```diff
@@ -15,6 +15,10 @@ export default {
     })
     return flag ? str : ''
   },
+  
+  isObject (val) {
+    return val !== null && typeof val === 'object' && !Array.isArray(val)
```
Comment (djhvscf) [inline src/utils/index.js:20]: Could we use `Object.prototype.toString.call(val) === '[object Object]'`?

Comment (UtechtDustin) [reply src/utils/index.js:20]: Maybe `val instanceof Object` ? :-)

File: src/bootstrap-table.js [hunk 2]
```diff
@@ -1454,7 +1454,8 @@ class BootstrapTable {
         type = column.radio ? 'radio' : type
 
         const c = column['class'] || ''
-        const isChecked = (value === true || value_ || (value && value.checked)) && value !== false
+        const isChecked = Utils.isObject(value) && value.hasOwnProperty('checked') ?
+          value.checked : (value === true || value_) && value !== false
```
[the remark this answers is not in the document]
Comment (TantumLabs) [reply src/bootstrap-table.js:1458]: That worked!

**Comment by UtechtDustin** [comment]
@TantumLabs Please check the conversations!
@djhvscf asked something, so your answer "it works" makes no sense, we dont only want working code, we also want good code!
Also you ignored my suggestion for the `isObject` function.

File: src/utils/index.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ export default {
   },
   
   isObject (val) {
-    return val !== null && typeof val === 'object' && !Array.isArray(val)
+    return val instanceof Object && !Array.isArray(val)
```
Comment (UtechtDustin) [inline src/utils/index.js:20]: if `val` is a `object`, its obviously not an `Array`.
So the second check can be removed.

File: src/bootstrap-table.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (djhvscf) [reply src/bootstrap-table.js:1458]: What do you mean? Did you test my suggestion?

File: src/utils/index.js (hunk 3 above)
Comment (djhvscf) [reply src/utils/index.js:20]: @TantumLabs  any news?

**Comment by djhvscf** [comment]
@TantumLabs any update here?

File: src/utils/index.js (hunk 3 above)
Comment (wenzhixin) [reply src/utils/index.js:20]: It's OK: https://live.bootstrap-table.com/code/wenzhixin/4670

File: src/bootstrap-table.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (wenzhixin) [reply src/bootstrap-table.js:1474]: We cannot use `value || value_` instead, because the `value` maybe `undefined`.

**Review by wenzhixin — APPROVED** [review]

**Pull request merged**